src/main/scala/org/dbpedia/spotlight/wikistats/JsonPediaParser.scala

```diff
@@ -59,5 +72,83 @@ class JsonPediaParser(lang:String)(implicit val sc: SparkContext,implicit val sq
   }
 
 
+  /*
+     Method to Get the list of Surface forms from the wiki
+   */
+  def getSfs() : RDD[String] = {
+
+    dfWikiRDD.select("wid","links.description")
+             .rdd
+             .map(artRow => (artRow.getList[String](1)))
+             .flatMap(sf => sf)
+  }
+
+  /*
+  Method to get the wid and article text from the wiki dump
+   */
+  //TODO Need to improve the filter condition of blank articles. we can include in the where clause of dataframe
+  def getArticleText(): RDD[(Long,String)] = {
+
+    dfWikiRDD.select("wid","wikiText")
+             .rdd
+             .map(artRow => {
+                 (artRow.getLong(0),artRow.getString(1))
+             })
+             .filter(artRow => artRow._2.length > 0)
+  }
+
+  /*
+   Logic to Get Surface Forms and URIs from the wikiDump
+   */
+  def getSfURI(): RDD[(Long,String,String)]= {
+
+    val sfUriRDD = dfWikiRDD.select("wid","links.description","links.id")
+             .rdd
+             .map(artRow => (artRow.getLong(0),artRow.getList[String](1),artRow.getList[String](2)))
+             .map{case (wid,sfArray,uriArray) => (wid,sfArray.zip(uriArray))}
+             .flatMap{case (wid,uriSf) => for(x <- uriSf) yield (wid,x._1,x._2)}
+    sfUriRDD
+
+  }
+  /*
+
+  Logic to Create Memory Token Store
+   */
+  def createTokenTypeStore(tokenTypes:List[TokenType]): MemoryTokenTypeStore =  {
```

Comment:
This should go on a companion object too, since it doesn't require any internal data. Maybe you can put it on the same object that the tokenizer is on and call it something like TextUtil?